ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -348,24 +369,37 @@ ledgerPeersThread inRng toPeerAddr tracer readUseLedgerAfter LedgerPeersConsensu
         let age = diffTime now oldTs
         (peerMap', bigPeerMap', ts) <- if age > peerListLifeTime
                              then
-                                 case useLedgerAfter of
-                                   DontUseLedger -> do
+                                 case useLedgerPeers of
+                                   DontUseLedgerPeers -> do
                                      traceWith tracer DisabledLedgerPeers
                                      return (Map.empty, Map.empty, now)
-                                   UseLedgerAfter slot -> do
-                                     peers_m <- atomically $ lpGetPeers slot
-                                     let peers    = maybe Map.empty accPoolStake peers_m
-                                         bigPeers = maybe Map.empty accBigPoolStake peers_m
-                                     traceWith tracer $ FetchingNewLedgerState (Map.size peers) (Map.size bigPeers)
-                                     return (peers, bigPeers, now)
-
+                                   UseLedgerPeers DontUseLedgerAfter -> do
+                                     peers <- (\case
+                                                BeforeSlot          -> []
+                                                LedgerPeers _ peers -> peers
+                                              )
+                                          <$> atomically (getLedgerPeers ledgerPeersConsensusInterface 1)
+                                     let peers' = accPoolStake peers
+                                         bigPeers = accBigPoolStake peers
+                                     traceWith tracer $ FetchingNewLedgerState (Map.size peers') (Map.size bigPeers)
+                                     return (peers', bigPeers, now)
+                                   UseLedgerPeers (UseLedgerAfter slot) -> do
+                                     peers <- (\case
+                                                BeforeSlot          -> []
+                                                LedgerPeers _ peers -> peers
+                                              )
+                                          <$> atomically (getLedgerPeers ledgerPeersConsensusInterface slot)
+                                     let peers' = accPoolStake peers
+                                         bigPeers = accBigPoolStake peers
+                                     traceWith tracer $ FetchingNewLedgerState (Map.size peers') (Map.size bigPeers)
+                                     return (peers', bigPeers, now)
```

Comment:
I think we can unify these two cases.  The only difference is the `slotNo` passed to `getLedgerPeers`, isn't it?